### PR TITLE
[DebugInfo][RemoveDIs] Prevent duplicate DPValues from being returned by findDbgIntrinsics

### PR DESCRIPTION
**Pull request by SLTozer**
(Probably) fixes the error described here: https://github.com/llvm/llvm-project/commit/a93a4ec7dd205b965ee5597314bb376520cd736c#commitcomment-138965199

The function `findDbgIntrinsics` is used to return a list of debug intrinsics and DPValues that use a given value, with the intent that no duplicates are returned in either list. For DPValues, we've guarded against DPValues that use a value multiple times as part of a DIArgList, but we have not guarded against DPValues that use a value multiple times as separate operands (currently only possible for `dbg_assign`s, something I missed in my implementation of that type!). This patch adds a guard, and also updates a test to cover this case.

NB: I think that the llvm.dbg.assign in the test isn't having its address expression updated correctly by CoroFrame, which may be a (completely unrelated) error; something to look at later, but not relevant to this fix.

### CODE REVIEW

**Comment by SLTozer** [comment]
> findDbgIntrinsics can be a hot path -- would you be able to pump this through the compile-time-tracker please?

http://llvm-compile-time-tracker.com/compare.php?from=ad49fe3e89c3b3950956548f14cdb5c159ba0aec&to=5fc56961c20dd877798e0158e7424ab7907a645a&stat=instructions:u

tl;dr this is a nothing for performance.

**Review by OCHyams — APPROVED** [review]
LGTM

> FIXME: Should we be updating the addresses on assigns here as well?

Agreed (but should be fixed separately)
